SpatialGDK/Source/SpatialGDK/Public/Tests/SpatialView/TestWorker.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialView/ViewCoordinator.h"
+#include "TestConnectionHandler.h"
+
+namespace SpatialGDK
+{
+class FTestWorker
+{
+public:
+	explicit FTestWorker(TArray<Worker_ComponentSetId> ComponentSetIds, FSendMessageCallback InSendMessageCallback = FSendMessageCallback(),
+						 FString InWorkerId = {}, Worker_EntityId InWorkerSystemEntityId = 0)
+	{
+		TUniquePtr<FTestConnectionHandler> Handler =
+			MakeUnique<FTestConnectionHandler>(InSendMessageCallback, InWorkerId, InWorkerSystemEntityId);
+		ConnectionHandler = Handler.Get();
```

Comment:
How does this pointer not get invalidated when Handler is destructed on constructor exit?

Reply:
Because the unique pointer is moved into the coordinator which guarantees it's life. I agree this pattern is odd though.

Although it's kind of why this class exists. It feels weirder in the normal cases to have a shared pointer passed into the coordinator as really doing this is in some way "unsafe". I think that the coordinator should have a function to return a raw pointer to the handler.

Reply:
Thinking about this more this is basically the only case I've come up with where you want to do weird things with the lifetime of this and all ways to make it safer involve adding a lot of structure that might not quite be right and makes other code less clear. And it's not clear what that structure should be.

Which was kind of the reason for making a TestWorker type that just encapsulates the weird bit so that the outward facing types all behave well.